content/shared/influxdb3-processing-engine.md

```diff
@@ -0,0 +1,84 @@
+# Processing engine
+
+The Processing engine is an embedded Python virtual machine that runs inside an {{% product-name %}} database server. It executes Python code in response to triggers and database events without requiring external application servers or middleware.
+
+## How it works
+
+### Architecture
+
+The Processing engine runs Python code directly within a {{% product-name %}} server process. This design provides high performance and direct access to database resources.
+
+- **Embedded execution**: Code runs in the same process space as the database server
+- **Direct data access**: Zero-copy access to data
+- **Event-driven**: Responds to database writes, scheduled events, and HTTP requests
+- **Isolated contexts**: Maintains separation between different plugin executions
+- **Cache integration**: Access to system caches including Last values  and Distinct values 
+
+### Event processing flow
+
+When events occur in the database, the Processing engine handles them through a consistent sequence:
+
+1. A **trigger** activates the plugin based on one of three event types:
```

Comment:
```suggestion
1. A  _trigger_ configures execution parameters and executes the plugin based on one of three event types:
```

Italics is preferred over bold for introducing terms. However, in this case, we should link them to a definition (not sure where that is at the moment).